docs/assignment3.md

```diff
@@ -1 +1,418 @@
+    * returns instance of the GistsApi singleton.
+* _setAuthorization(String providedAuthorization): Void_
+    * set the authorization token to the one the users has provided on login.
+* _setUsesGithub(boolean providedUsesGithub): Void_
+    * boolean that shows wether or not the user is using github gists integration
+* _getUsesGithubGists(): Boolean_
+    * getter function for the usesGithub variable
+* _getAllSnippets(): List<Snippet>_
+    * returns all the gists and parses them to match our snippets
+* _getSpecificSnippet(String gistsId): Snippet_
+    * returns a specific gist with passed gistId
+* _postSpecificSnippet(Snippet snippet): String_
+    * posts a snippet to gist
+* _patchSpecificSnippet(Snippet snippet): Void_
+    * updates a specific snippet to gist
+* _deleteSpecificSnippet(String gistId): Void_
+    * deletes a specific snippet from gist
+* _sendHttpRequest(HttpRequest request):HttpResponse<String>
+    * send a http request, used to send the request to the Gists api.
+* _buildHttpRequest(String method, String url, String body): HttpRequest_
+    * build a http request, used to build the reqeusts we send to the Gists api
+* _onException(Exception e): Void_
+    * handles errors.
+#### Associations
+* GistsApi > SnippetManager 
+    *  provides api.
+* GistsApi > snippet
+    * provides api.
+* GistsApi > JSonIo
+    * provides api.
+## Object diagrams								
+Author(s): Yael Goede
+
+![snapshot Diagram](snapshotDiagram.png)
+In the diagram above a snapshot of the system is shown where the database class is filled with 4 snippets. All snippets have a unique id which makes them unique from each other. Further more all fields are filled with the required information, such as the language, tags and content field. 
+The content field contains the actual code used as snippet by the user. The language field specifies the programming language which is valid for the content field. This field enables features such as the text highlighting feature, which is language specific.
+
+## State machine diagrams									
+Author(s): `Serghei`
+
+![State Machine CLI Diagram](State_Machine_Cli.svg)
+- Class: CliUI
+- CliUI is the entry point of the application. It provides an user interface to the functionality of the program and is concerned with getting all of the user input except for the editor window and gists data. Moreover, CliUI also contains the logging in functionality (because it is too simple to have a class on its own). At the start of the app, the CliUI asks the user if the user would like to authorize and, if yes, in which way: by using their Github access token or Basic auth. Then, depending on the option selected, CliUI will set the relevant static variables as necessary and proceed. If the app was called using command-line arguments, it will execute the respective command and exit. Otherwise (the more common use case), the UI loop will start: dispay menu, get command, execute command, repeat until the command is QUIT.
+
```

Comment:
-At the not logged in state, you could've had a decision node to indicate the different paths taken based on input.
-Log-in the composite state in itself could've had an entry activity to ask for credentials such as password.
-Good otherwise